File: lab4/lang-tests/test-4/scope.c
#include <stdio.h>

int global = 1;

int function_a(int z){
    printf("en function_a 'z' vale: %d\n",z);
    printf("en function_a 'global' vale: %d\n",global);
    printf("como 'global' vale 4 entonces tomo el valor del bloque en function_b\n");
    printf("entonces C tiene un ALCANCE ESTATICO\n");
    return z + global;
}

int function_b(int y){
    printf("\nen function_b 'y' vale: %d\n",y);
    int global = y+1;
    printf("'global' es local para function_b y vale: %d\n\n",global);
    return (function_a(global*y));
}

int main(void){
    int res = 0;
    res = function_b(3);
    printf("%d\n\n",res);

    return 0;
}

File: lab4/lang-tests/test-4/scope.hs
import System.IO

y = 1
function_a :: Integer -> IO Integer
function_a z = 
    do
        print("el valor de 'z' en function_a es: ")
        print(z)
        print("el valor de 'y' en function_a es: ")
        print(y)
        return z


function_b :: Integer -> IO Integer
function_b z =
    do
        let y = 3
        res <- function_a(y+z)
        return res


main =
    do
        let y = 2
        function_b y